python/cugraph/cugraph/utilities/utils.py

```diff
@@ -502,3 +503,25 @@ def sample_groups(df, by, n_samples):
     result = df.loc[df.groupby(by)["_"].rank("first") <= n_samples, :]
     del result["_"]
     return result
+
+
+def create_list_series_from_2d_ar(ar, index):
+    """
+    Create a cudf list series  from 2d arrays
+    """
+    n_rows, n_cols = ar.shape
+    data = as_column(ar.flatten())
+    offset_col = as_column(
+        cp.arange(start=0, stop=len(data) + 1, step=n_cols), dtype="int32"
+    )
+    mask_col = cp.full(shape=n_rows, fill_value=True)
+    mask = cudf._lib.transform.bools_to_mask(as_column(mask_col))
```

Comment:
Is this intended to be a public API, or is there a risk cuDF could change it and break us?

Reply:
There is a risk that cuDF could change it later on but I could not come up with something without it. 

https://github.com/search?q=repo%3Arapidsai%2Fcudf+bools_to_mask+language%3APython&type=code&l=Python